lasagne/layers/cuda_convnet.py

```diff
@@ -29,17 +29,177 @@
+        is disabled by default. In most cases this does not make a difference
+        anyway because the filters are learnt. However, ``flip_filters`` should
+        be set to ``True`` if weights are loaded into it that were learnt using
+        a regular :class:`lasagne.layers.Conv2DLayer`, for example.
+
+    partial_sum : int or None (default: 1)
+        This value tunes the trade-off between memory usage and performance.
+        You can specify any positive integer that is a divisor of the output
+        feature map size (i.e. output height times output width). Higher
+        values decrease memory usage, but also performance. Specifying 0 or
+        ``None`` means the highest possible value will be used. The Lasagne
+        default of ``1`` gives the best performance, but also the highest
+        memory usage.
+
+        More information about this parameter can be found in the
+        `cuda-convnet documentation
+        <https://code.google.com/p/cuda-convnet/wiki/LayerParams>`_.
+
+    **kwargs
+        Any additional keyword arguments are passed to the `Layer` superclass.
+
+    Attributes
+    ----------
+    W : Theano shared variable
+        Variable representing the filter weights.
+
+    b : Theano shared variable
+        Variable representing the biases.
+
+    Notes
+    -----
+    Unlike :class:`lasagne.layers.Conv2DLayer`, this layer properly supports
+    the 'same' border mode. It is not emulated. This should result in better
+    performance.
+
+    Only one of ``pad`` and ``border_mode`` should be specified.
+
+    The cuda-convnet convolution implementation has several limitations:
+
+    * only square filters are supported.
+    * only identical strides in the horizontal and vertical direction are
+      supported.
+    * the number of filters must be a multiple of 16.
+    * the number of input channels must be even, or less than or equal to
+      3.
+    * if the gradient w.r.t. the input is to be computed, the number of
+      channels should be divisible by 4.
+    * performance is optimal when the batch size is a multiple of 128 (but
+      other batch sizes are supported).
+    * this layer only works on the GPU.
+
+    The cuda-convnet convolution implementation uses c01b (batch-size-last)
+    axis order by default. The Theano/Lasagne default is bc01
+    (batch-size-first). This layer automatically adds the necessary dimshuffle
+    operations for the input and the parameters so that it is interoperable
+    with other layers that assume bc01 axis order.
+
+    However, these additional dimshuffle operations may sometimes negatively
+    affect performance. For this reason, it is possible to disable them by
+    setting ``dimshuffle=False``.
+
+    In this case, the user is expected to manually ensure that the input and
+    parameters have the correct axis order. :class:`ShuffleBC01ToC01BLayer` and
+    :class:`ShuffleC01BToBC01Layer` can be used to convert between bc01 and
+    c01b axis order.
```

Comment:
It's a bit confusing that these are three separate paragraphs although they're all about the same issue. Does it look bad in Sphinx (worse than usual, I mean ;)) if you remove the empty lines in between?